### PR TITLE
Enable USD version check for components distributed separately

### PR DESCRIPTION
**Pull request by kxl-adsk**
USD plugins that are distributed in separate packages should perform runtime USD version check to avoid potential crashes caused by version mismatch. This PR adds:
- `MAYA_USD_VERSION`  and `MAYAUSD_VERSION` env variables via MayaUSD mod file 
- `MAYA_PXR_PLUGINPATH_NAME` env variable and a service that will append `<USD_VERSION>` to each path and call RegisterPlugins for them. This service runs only once per Maya session and is triggered by either MayaUSD plugin or mtoh (whichever is loaded first). For example, MayaUSD distribution with USD 20.11 will attempt registration of each path in MAYA_PXR_PLUGINPATH_NAME with the `2011` subdirectory.

### CODE REVIEW

**Comment by kxl-adsk** [comment]
Still need to add a test, but wanted to open it to collect feedback.

File: lib/mayaUsd/nodes/proxyShapePlugin.cpp [hunk 1]
```diff
@@ -62,6 +63,9 @@ MayaUsdProxyShapePlugin::initialize(MFnPlugin& plugin)
         return MS::kSuccess;
     }
 
+    // Call one time registration of plugins compiled for same USD version as MayaUSD plugin.
+    MAYAUSD_NS::registerVersionedPlugins();
```
Comment (seando-adsk) [inline lib/mayaUsd/nodes/proxyShapePlugin.cpp:67]: Here in the core, you should just use MayaUsd::

File: lib/mayaUsd/render/mayaToHydra/plugin.cpp [hunk 2]
```diff
@@ -34,6 +36,9 @@ PXR_NAMESPACE_USING_DIRECTIVE
 PLUGIN_EXPORT MStatus initializePlugin(MObject obj) {
     MStatus ret = MS::kSuccess;
 
+    // Call one time registration of plugins compiled for same USD version as MayaUSD plugin.
+    MAYAUSD_NS::registerVersionedPlugins();
```
Comment (seando-adsk) [inline lib/mayaUsd/render/mayaToHydra/plugin.cpp:40]: Same

File: lib/mayaUsd/utils/plugRegistryHelper.h [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+//
+#ifndef MAYAUSD_PLUGREGISTRYHELPER_H
+#define MAYAUSD_PLUGREGISTRYHELPER_H
+
+#include <mayaUsd/base/api.h>
+
+#include <pxr/base/plug/registry.h>
+#include <pxr/base/tf/getenv.h>
+#include <pxr/base/tf/pathUtils.h>
+#include "pxr/base/tf/stringUtils.h"
```
Comment (seando-adsk) [inline lib/mayaUsd/utils/plugRegistryHelper.h:24]: Why does this one have " and not <

Comment (kxl-adsk) [reply lib/mayaUsd/utils/plugRegistryHelper.h:24]: Copy-paste error, thx!

File: lib/mayaUsd/nodes/proxyShapePlugin.cpp (hunk 1 above)
Comment (kxl-adsk) [reply lib/mayaUsd/nodes/proxyShapePlugin.cpp]: As per offline discussion, I removed all usages of MAYAUSD_NS.

**Review by seando-adsk — APPROVED** [review]

**Review by seando-adsk — CHANGES_REQUESTED** [review]
Will you fix the " vs < in plugRegistryHelper.h?

**Review by seando-adsk — APPROVED** [review]

**Pull request merged**